source/platform/include/Gwork/Renderers/Allegro5.h

```diff
@@ -111,6 +112,8 @@ namespace Gwk
 
             std::unordered_map<Font, ALFontData> m_fonts;
             std::unordered_map<Texture, ALTextureData> m_textures;
+            std::pair<const Font, ALFontData>* m_lastFont;
```

Comment:
Why store the last font here? Why not store an opaque handle in `Font`, like it was? If the font keeps changing you have to constantly look it up. The only cache you need is to look up the font by family/size, etc, i.e. style.

Reply:
The reason for avoiding the user data in Font/Texture primarily is to remove data specific to a renderer having the potential to be shared between renderers. One of my use cases for Gwork is an environment where multiple renderers exist. Some of the Gwork objects are shared between these renderers. The user data in Font/Texture does not have a way to determine the renderer that assigned the data or store multiple without recreating every time the renderers change. 

I also question the lifespan of user data in Font/Texture.

Reply:
Multiple renderers?! I'm curious, how so? Multiple instances with the same backend, or multiple backends?

Reply:
It has always been assumed that an application would embed one renderer. I have thought about having fallback renderers if some features aren't available on other renderers, e.g. if more primitives were added to add a canvas (e.g. drawing of charts etc) it call fallback to software renderer if a particular renderer did not have that features. It's tempting to include something like [NanoVG](https://github.com/memononen/nanovg) to do something like this.

Reply:
One example of multiple renderers is in-game overlays. Without knowing which rendering API the game will you, multiple backends are used. However, some game when switching between lobby & battle(for lack of a better word), different rendering APIs are used. Usually different versions of Direct X. And in rare cases, the HUD is rendered in DX 10 and the 3D environment in DX 11.

Reply:
Ok, this has never been a requirement before. Interesting. I was wondering why the extensive changes and whether is was overkill! I can see your motivation now!

Reply:
I added #87 so this feature can be mentioned as it is significant. Will have to mention/update docs.